lib/babe/crypto.go

```diff
@@ -84,6 +84,9 @@ func checkPrimaryThreshold(randomness Randomness,
 // equation: threshold = 2^128 * (1 - (1-c)^(1/len(authorities))
 // see https://github.com/paritytech/substrate/blob/master/client/consensus/babe/src/authorship.rs#L44
 func CalculateThreshold(C1, C2 uint64, numAuths int) (*scale.Uint128, error) {
+	if C1 == 0 || C2 == 0 {
+		return nil, errors.New("invalid input: C1 and C2 cannot be 0")
```

Comment:
make this a global exported error i.e.

```go
var ErrThresholdBothZero = errors.New("both inputs cannot be zero")
```

and have it as

```suggestion
		return nil, ErrThresholdBothZero
```

Also, unless `C1` and `C2` have an actual naming meaning, consider changing them to `c1` and `c2` and don't mention them in the error message.

Reply:
the error should say `numerator or denominator cannot be 0` or something like that, as C1 and C2 represent the numerator and denominator to `c` in the equation in the comment, and neither should be 0